lib/src/main/java/com/cloudbees/groovy/cps/impl/BlockScopeEnv.java

```diff
@@ -24,8 +27,9 @@ public BlockScopeEnv(Env parent) {
     public BlockScopeEnv(Env parent, int localsSize) {
         super(parent);
         if (localsSize <= 0) {
-            locals = Collections.emptyMap();
-            types = Collections.emptyMap();
+            // Lazily declare using EMPTY_MAP to conserve memory until we actually declare some variables
+            locals = Collections.EMPTY_MAP;
```

Comment:
If you wanted, you could avoid an `unchecked`/`rawtypes` warning by going back to `emptyMap()` but switching the test from

```java
locals == Collections.EMPTY_MAP
```

(theoretically unsafe) to

```java
!(locals instanceof HashMap)
```

(definitely safe).